test/inductor/test_aot_inductor.py

```diff
@@ -1268,6 +1269,50 @@ def forward(self, x):
         example_inputs = (torch.randn(4, device=self.device),)
         self.check_model(Model(), example_inputs)
 
+    def test_triton_kernel_with_none_input(self):
+        if self.device != "cuda":
+            raise unittest.SkipTest("requires CUDA")
+
+        class Model(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x, y):
+                # AOT export does not allow for input mutation
+                n_elements = x.size()[0]
+                BLOCK_SIZE = 1024
+
+                x = x.clone()
+                y = y.clone()
+                output_wo_y = torch.empty_like(x)
+                output_with_y = torch.empty_like(x)
+
+                wo_kernel = add_kernel_with_optional_param[(1,)](
+                    x,
+                    None,
```

Comment:
I thought the problem was it was being passed a tensor first and then later None?

Reply:
This ended up being more subtle than we thought. Tensor arguments can be `None` or not `None`, bu (I believe) checking if `in_ptr1 is None` in the Triton kernel code doesn't quite work: the generated PTX code doesn't seem to be affected by this. Instead, there should be another, typically a `tl.constexpr` argument that decides the control flow, set in accordance with the respective tensor arguments being set to `None` or not. As a result, your existing de-duping mechanism doesn't dedup the calls with `None` and not, due to another constant arg getting different values.

Reply:
Ok, so de-duping mechanism is correct then

Reply:
It's not necessarily guarded against None / non-None tensor args, but those are normally accompanied by constexpr args with different values, so should be fine.